modules/math/src/main/java/com/opengamma/strata/math/impl/interpolation/CubicSplineInterpolator.java

```diff
@@ -47,16 +50,8 @@ public PiecewisePolynomialResult interpolate(final double[] xValues, final doubl
       ArgChecker.isFalse(Double.isInfinite(yValues[i]), "yData containing Infinity");
     }
 
-    for (int i = 0; i < nDataPts; ++i) {
-      for (int j = i + 1; j < nDataPts; ++j) {
-        ArgChecker.isFalse(xValues[i] == xValues[j], "Data should be distinct");
-      }
-    }
-
-    double[] xValuesSrt = new double[nDataPts];
-    double[] yValuesSrt = new double[nDataPts];
-
-    xValuesSrt = Arrays.copyOf(xValues, nDataPts);
+    double[] xValuesSrt = Arrays.copyOf(xValues, nDataPts);
```

Comment:
I suspect `xValues.clone()` might be quicker in cases like this.

Reply:
Apparently `.clone()` may add a cast (since it's implicitly overriden from the `Object::clone()` method)
`Arrays.copyOf()` delegates to `System.arrayCopy` which is native, and doesn't involve a cast.
I'd have to benchmark to be sure